pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1001,6 +1001,7 @@ type httpListenerOpts struct {
 
 	suppressEnvoyDebugHeaders bool
 	skipIstioMXHeaders        bool
+	mtlsGatewayMxFilter       bool
```

Comment:
Comment ? I hope this will only be for Egress and Waypoint gateways ( internal ) - and never for the Ingress gateway.

Reply:
Pretty sure it applies to ingress as well, if they set ISTIO_MUTUAL, no?

There is no "ingress" or "egress" in any pilot code AFAIK, so there cannot be this distinction (unless we change that)

Reply:
Yes, As mentioned here: https://github.com/istio/istio/pull/48141#issuecomment-1838969823.

This change is currently applied to Ingress and Egress gateways.